src/types/error.js

```diff
@@ -264,19 +264,45 @@ export class SwapCurrencyError extends Error {
   +pluginId: string
   +fromCurrency: string
   +toCurrency: string
+  +fromTokenId: string | void
+  +toTokenId: string | void
 
   constructor(
     swapInfo: EdgeSwapInfo,
-    fromCurrency: string,
-    toCurrency: string
+    // Passing currency codes is deprecated:
+    request: string | EdgeSwapRequest,
+    toCurrency?: string
   ) {
+    // Backwards-compatible currency code extraction:
+    let fromCurrency: string
+    if (typeof request === 'string') {
+      fromCurrency = request
+      if (toCurrency == null) toCurrency = 'unknown'
+    } else {
+      fromCurrency =
+        request.fromTokenId != null
+          ? request.fromWallet.currencyConfig.allTokens[request.fromTokenId]
+              .currencyCode
+          : request.fromCurrencyCode ??
+            request.fromWallet.currencyInfo.currencyCode
+      toCurrency =
+        request.toTokenId != null
+          ? request.toWallet.currencyConfig.allTokens[request.toTokenId]
+              .currencyCode
+          : request.toCurrencyCode ?? request.toWallet.currencyInfo.currencyCode
+    }
+
     super(
       `${swapInfo.displayName} does not support ${fromCurrency} to ${toCurrency}`
```

Comment:
Should probably show the tokenIds as well to differentiate what was trying to be used. That's the purpose of the error.

Reply:
The error is meant to be translated by the GUI into something more usable, which is why we put the tokenId's into the data part. I can adjust the message to include those as well, in case the GUI doesn't do the translation.

Reply:
That would be great. Also helps for the logs that gets spewed.